client/client.go

```diff
@@ -286,98 +286,106 @@ func isAllRegions(regions []string) bool {
 
 }
 
+func getAccountId(ctx context.Context, awsCfg aws.Config) (*sts.GetCallerIdentityOutput, error) {
+	svc := sts.NewFromConfig(awsCfg)
+	return svc.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{}, func(o *sts.Options) {
+		o.Region = "aws-global"
+	})
+
+}
+
+type AssumeRoleAPIClient interface {
+	AssumeRole(ctx context.Context, params *sts.AssumeRoleInput, optFns ...func(*sts.Options)) (*sts.AssumeRoleOutput, error)
+}
+
+func configureAwsClient(ctx context.Context, logger hclog.Logger, awsConfig *Config, account Account, stsClient AssumeRoleAPIClient) (aws.Config, error) {
+	var err error
+	var awsCfg aws.Config
+	configFns := []func(*config.LoadOptions) error{
+		config.WithDefaultRegion(defaultRegion),
+		config.WithRetryer(newRetryer(awsConfig.MaxRetries, awsConfig.MaxBackoff)),
+	}
+
+	if account.LocalProfile != "" {
+		configFns = append(configFns, config.WithSharedConfigProfile(account.LocalProfile))
+	}
+
+	awsCfg, err = config.LoadDefaultConfig(ctx, configFns...)
+
+	if err != nil {
+		return awsCfg, fmt.Errorf(awsFailedToConfigureErrMsg, account.accountName, err, checkEnvVariables())
+	}
+
+	if account.RoleARN != "" {
+		opts := make([]func(*stscreds.AssumeRoleOptions), 0, 1)
+		if account.ExternalID != "" {
+			opts = append(opts, func(opts *stscreds.AssumeRoleOptions) {
+				opts.ExternalID = &account.ExternalID
+			})
+		}
+		if stsClient == nil {
+			stsClient = sts.NewFromConfig(awsCfg)
+		}
+		provider := stscreds.NewAssumeRoleProvider(stsClient, account.RoleARN, opts...)
+		// Test out retrieving credentials
+
+		awsCfg.Credentials = aws.NewCredentialsCache(provider)
+	}
+
+	if awsConfig.AWSDebug {
+		awsCfg.ClientLogMode = aws.LogRequest | aws.LogResponse | aws.LogRetries
+		awsCfg.Logger = AwsLogger{logger.With("accountName", account.accountName)}
```

Comment:
We don't have the account id at this time we only get the account id after we setup the aws client and call `get-caller-identity` 


Part of this pr is to remove the accountid field because it was overloaded in meaning.